tests/std/tests/LWG2597_complex_branch_cut/test.cpp

```diff
@@ -0,0 +1,168 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <cassert>
+#include <cmath>
+#include <complex>
+#include <limits>
+
+using namespace std;
+
+template <typename T>
+void check(const T x, const T y) {
+    constexpr T tolerance = 2 * numeric_limits<T>::epsilon();
+
+    if (isfinite(x) && isfinite(y)) {
+        assert(signbit(x) == signbit(y));
+        assert(abs(x - y) <= tolerance * fmax(abs(x), abs(y)));
+    } else if (isnan(x) || isnan(y)) {
+        assert(isnan(x) == isnan(y));
+    } else {
+        assert(x == y);
+    }
+}
+
+template <typename T>
+void check_part(const complex<T>& x, const complex<T>& y) {
+    check(x.real(), y.real());
+    check(x.imag(), y.imag());
+}
+
+template <typename T>
+void check_norm(const complex<T>& x, const complex<T>& y) {
+    constexpr T tolerance = 4 * numeric_limits<T>::epsilon();
+
+    if (isfinite(x.real()) && isfinite(x.imag()) && isfinite(y.real()) && isfinite(y.imag())) {

```

Comment:
`check_norm(finite, NaN)` doesn't fire an assertion as expected.

```suggestion
    const bool x_is_finite = isfinite(x.real()) && isfinite(x.imag());
    const bool y_is_finite = isfinite(y.real()) && isfinite(y.imag());
    assert(x_is_finite == y_is_finite);

    if (x_is_finite) {
```